### DISCUSSION TITLE
Pendulum 3.0.0 broke DBAPI interaction

### DISCUSSION THREAD

**Discussion by rishi-kulkarni** [apache/airflow#37037] [General]
### Apache Airflow version

2.8.1

### What happened?

My team uses the provided SQLiteHook to store some data on disk between tasks. We timestamp the column by creating some pandas DataFrame and inserting into the SQLite file with the hook. 

```
    df = df.rename(columns=str.lower).astype({"user_id": int}).assign(date_created=data_interval_end)
    sqlite_hook.insert_rows(
        failure_event_sqlite_table,
        list(df[["user_id", "shift_id", "successes", "failures", "date_created"]].itertuples(index=False)),
    )

```

This has (and still does) make rows that look like this:

```
   user_id  shift_id  successes  failures               date_created
0        1        11        1.0       0.1  2023-09-25T00:00:00+00:00
```

In later tasks, we may pull out some data with a query that looks like this:
```
sqlite_hook.get_pandas_df(
        f"select * from {failure_event_sqlite_table} where date_created = ?", parameters=(data_interval_end,)
    )
```
This used to generate a query with a timestamp in the same format, but now it generates a query with a timestamp that looks like this (note the missing `T`)
```
where date_created = '2023-07-04 18:00:00+00:00'
```

This is inconsistent behavior breaks all of these queries - the pendulum timestamps should serialize and deserialize in the same format. 

### What you think should happen instead?

The previous behavior was to make statements like this:

```
where date_created = '2023-07-04T18:00:00+00:00'
```
But it's fine either way, as long as it's consistent. 

### How to reproduce

```
@task
def demonstrate_broken_sqlite(
    *,
    data_interval_end: "Optional[DateTime]" = None,
):
    import pandas as pd
    from airflow.providers.sqlite.hooks.sqlite import SqliteHook

    assert data_interval_end is not None

    sqlite_hook = SqliteHook()

    df = pd.DataFrame(
        {
            "user_id": [1, 2, 3],
        }
    ).assign(date_created=data_interval_end)

    df = df.rename(columns=str.lower).astype({"user_id": int})

    sqlite_hook.run(
        """
        create table if not exists test_table (
            user_id integer primary key,
            date_created datetime        
        )
        """
    )

    sqlite_hook.insert_rows(
        "test_table",
        list(df[["user_id", "date_created"]].itertuples(index=False)),
    )

    out = sqlite_hook.get_pandas_df("select * from test_table where date_created = ?", parameters=[data_interval_end])

    assert len(out) == 3 # fails
```

### Operating System

AmazonLinux2023

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
sure - pendulum 3 is not compatible with Pendulum 2 - it broke Airflow in a feae places. So it also broke interaction with DBAPI/pands. most likely - and since we are passing the queries directly to dbapi, you should likely raise the issue to Pendulum or sqlite dbapi, there is not much we can do in Airflow about it.

**Comment by potiuk** [comment]
Converted to discussion if needed

**Comment by Taragolis** [comment]
My 2 cents, I think that current behaviour with string interpretation is better in Pendulum 3, rather than in previous version because it work exactly the same as it work with `datetime.datetime`. I guess the problem might happen only for datebase without native date/datetime support, such as SQLite. SO it his case i would recommend:
1. Use explicitly [`.isoformat(sep='T')`](https://docs.python.org/3/library/datetime.html#datetime.date.isoformat)
2. Postpone use `.isoformat(sep='T')` and rather than install pendulum 2.1.2 which would also supported for a while
